syndesis/src/modules/connections/pages/ConnectionsPage.tsx

```diff
@@ -73,59 +73,69 @@ export default class ConnectionsPage extends ListViewToolbarAbstractComponent<
   public render() {
     return (
       <WithConnections>
-        {({ data, hasData, error }) => {
+        {({ read, data, hasData, error }) => {
           const filteredAndSortedConnections = getFilteredAndSortedConnections(
             data.items,
             this.state.activeFilters,
             this.state.currentSortType,
             this.state.isSortAscending
           );
+
           return (
-            <ConnectionsListView
-              linkToConnectionCreate={'/connections/create'}
-              filterTypes={filterTypes}
-              sortTypes={sortTypes}
-              resultsCount={filteredAndSortedConnections.length}
-              {...this.state}
-              onUpdateCurrentValue={this.onUpdateCurrentValue}
-              onValueKeyPress={this.onValueKeyPress}
-              onFilterAdded={this.onFilterAdded}
-              onSelectFilterType={this.onSelectFilterType}
-              onFilterValueSelected={this.onFilterValueSelected}
-              onRemoveFilter={this.onRemoveFilter}
-              onClearFilters={this.onClearFilters}
-              onToggleCurrentSortDirection={this.onToggleCurrentSortDirection}
-              onUpdateCurrentSortType={this.onUpdateCurrentSortType}
-            >
-              <ConnectionsGrid>
-                <WithLoader
-                  error={error}
-                  loading={!hasData}
-                  loaderChildren={
-                    <>
-                      {new Array(5).fill(0).map((_, index) => (
-                        <ConnectionsGridCell key={index}>
-                          <ConnectionSkeleton />
-                        </ConnectionsGridCell>
-                      ))}
-                    </>
+            <WithPolling read={read} polling={5000}>
```

Comment:
👍 

I think we could add a `WithUpdatingConnections` helper class that hides this away from the page and keeps the logic in the `@syndesis/api` package.

Reply:
Yeah, it feels like the page has to do a lot of complicated looking composition for sure.  I was meaning to update the pages that had WithPolling added to use the server event listener stuff, I'll work on adding these components instead and update these pages appropriately, thanks!

Reply:
Also please find a better name than `WithUpdatingConnections`, I suck at naming 😂

Reply:
Maybe I'll rename the existing WithConnections etc to something like 'WithConnectionsInternal' and keep export the helper class instead as WithConnections?

Reply:
What if we keep a single access point-eg. `WithConnections`-but add a property to enable the update strategy (that has been decided upfront by the component developer)?

So I can go and do `<WithConnection updates={true}>`, or `<WithConnection disableUpdates={true}>` if we want to have the auto-updating behavior on by default.

Reply:
As a side benefit, it will enable us to disable the updating mechanism while debugging using the React Dev Tools :)

Reply:
Yeah, that sounds good!